src/ui/RosterItem.ts

```diff
@@ -174,6 +174,7 @@ export default class RosterItem {
       let inputElement = this.element.find('input');
 
       this.contact.setName(<string> inputElement.val());
+      this.contact.getNicknameObject().setNickVisible(false);
```

Comment:
I guess this should enforce the user defined name. But this would be easier with priorities. E.g. `getName() || getNickname() || getBareJid()`.